Cython/Compiler/Code.py

```diff
@@ -2412,7 +2412,7 @@ def error_goto_if_null(self, cname, pos):
         return self.error_goto_if("!%s" % cname, pos)
 
     def error_goto_if_neg(self, cname, pos):
-        return self.error_goto_if("%s < 0" % cname, pos)
+        return self.error_goto_if("(%s < 0)" % cname, pos)
```

Comment:
This is worth a comment. Otherwise, the next review will end up removing it.
```suggestion
        # Add extra parentheses to silence clang warnings about constant conditions.
        return self.error_goto_if("(%s < 0)" % cname, pos)
```